docs/java/features/connectivity/destinations.md

```diff
@@ -18,6 +18,37 @@ The SAP Cloud SDK offers some basic functionality that helps with connecting to
 
 This concept is integrated with the [Destination Service](https://help.sap.com/viewer/cca91383641e40ffbe03bdc78f00f681/Cloud/en-US/7e306250e08340f89d6c103e28840f30.html) that is available on SAP Cloud Platform. If the application has a service binding to this service in place the SDK will provide access to these destinations.
 
+## Accessing All Destinations in Cloud Foundry ##
+ 
+ Inorder to fetch all destinations from Destination Service you need to do the following steps: 
+
+ 1) Invoke the `ScpCfDestinationLoader` with the `DestinationService` and `XsuaaService` instances.
+
+```java
+ final ScpCfDestinationLoader destinationLoader = new ScpCfDestinationLoader(destinationService, xsuaaService);
+```
+
+ 2) Build `DestinationOptions` object based on the retrieval strategy for your particular use case. Retieval strategy can be `ALWAYS_SUBSCRIBER`, `ALWAYS_PROVIDER` or `SUBSCRIBER_THEN_PROVIDER`. Refer the code below for better understanding:  
```

Comment:
I would like to know if this explanation on DestinationOptions object is sufficient? (i.e we need it here only to configure different retrieval strategies.) or I am missing something here?

Reply:
Honestly I am quite surprised that:

a) the default is always subscriber (that is not consistent with the default of `tryGetDestination` which is subscriber then provider)

and b) that we need this at all. I would have expected just `tryGetAllDestinations()` or `tryGetAllDestinations(RetrievalStrategy)`.

Reply:
> a) the default is always subscriber (that is not consistent with the default of tryGetDestination which is subscriber then provider)

I guess the default for `tryGetDestination ` is also always subscriber. I understood this by Line140 in `ScpCfDestinationLoader`.

> and b) that we need this at all. I would have expected just tryGetAllDestinations() or tryGetAllDestinations(RetrievalStrategy).

 I think we need this to set the Retrieval strategy. Hence,  you are right the API could have been just `tryGetAllDestinations(RetrievalStrategy).` . 

I would like to know what Alex think about this.

Reply:
> I guess the default for `tryGetDestination ` is also always subscriber. I understood this by Line140 in `ScpCfDestinationLoader`.

I would second that. This seems to be the case [since 3.0.0](https://github.wdf.sap.corp/MA/sdk/blob/rel/3.0.0/cloudplatform/cloudplatform-connectivity-scp-cf/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/ScpCfDestinationLoader.java#L103).

> > b) that we need this at all. I would have expected just tryGetAllDestinations() or tryGetAllDestinations(RetrievalStrategy).
> 
> I think we need this to set the Retrieval strategy. Hence, you are right the API could have been just `tryGetAllDestinations(RetrievalStrategy).`

Please note the rest of the API. The `RetrievalStrategy` is only exposed through `DestinationOptions`. You may oppose this decision now, but it was made from the beginning and changing that is not possible without breaking everything.

The loader classes, i.e. `ScpCfDestinationLoader`, are made public for **advanced** users. Convenience methods are to be exposed through `DestinationAccessor`. Since this feature is only available on CF, we cannot expose it on that general class. Hence we are talking about an advanced feature.

If you are challenging the API, please create a team-internal follow-up issue. Also include Chris, one major author of the Destinations API.

Reply:
I am ok with the current API.